internal: (studio) allow tests to be created in specs that have .only tests

astone123 opened the pull request:
<!-- Thanks for contributing! PLEASE...
- Read our contributing guidelines: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md 
- Read our Code Review Checklist on coding standards and what needs to be done before a PR can be merged: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#Code-Review-Checklist
- Mark this PR as "Draft" if it is not ready for review.
- Make sure you set the correct base branch based on what packages you're changing: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#branches
-->

- Closes https://github.com/cypress-io/cypress-services/issues/11209

### Additional details
<!-- Examples:
- Why was this change necessary?
- What is affected by this change?
- Any implementation details to explain?
-->

There's an issue when you try to create a new test in studio when the spec has a test that is marked as `only`. This PR fixes that by essentially ignoring `only` when we're in studio mode. When we're in studio, we always want to run the selected test regardless of if there's filtering going on in the spec. 

### Steps to test
<!--
For non-trivial behavior changes, list the steps that a reviewer should follow to validate the new behavior.
This is not meant to be the only testing performed by a reviewer, just the "happy path" that leads to the new behavior.
-->

Check out the e2e test I added - if you revert the changes in the runner you can see how it used to fail. Now you can create a new test and do everything else in studio even when the spec has an `only` test in it.

### PR Tasks
<!-- 
These tasks must be completed before a PR is merged.
If a task does not apply, write [na] instead of checking the box.
DO NOT DELETE the PR checklist.
-->

- [x] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? <!-- Link to PR here -->
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

astone123 committed:
internal: (studio) allow tests to be created in specs that have .only tests

astone123 committed:
fix stop-only

jennifer-shehane reviewed (changes requested):
@astone123 Whenever I click into this test with an .only on it to edit in Studio - it errors like it cannot detect a test. I think it may have to do with having the describe and a root test?

```js
describe('test project', () => {
  it.only('New test', function() {
    cy.visit('https://www.cypress.io')

  });
})
it('My new test', function() {

});
```

<img width="1335" height="580" alt="Image" src="https://github.com/user-attachments/assets/9836876d-2c4d-4ad3-97f0-0e8ea0c98cc0" />

mabela416 reviewed (changes requested):
If you create a new test and then try to reopen the test that has `.only`, the test that was created is what gets opened

https://github.com/user-attachments/assets/18779374-cddb-4f52-801f-c658af6b57c9


I used the `BranchSelection_spec.ts` file from the dashboard in cypress-services and I added `.only` to this [test](https://github.com/cypress-io/cypress-services/blob/develop/frontend/packages/dashboard/cypress/integration/branch-summary/Review/BranchSelection_spec.ts#L109)

jennifer-shehane committed:
Merge branch 'release/15.0.0' into studio-fix-only

astone123 committed:
add system test spec, add test, fix bugs

astone123 committed:
clean up unused logic

astone123 committed:
fix issue

astone123 committed:
another fix

astone123 committed:
Merge branch 'release/15.0.0' into studio-fix-only

astone123 commented:
> @astone123 Whenever I click into this test with an .only on it to edit in Studio - it errors like it cannot detect a test. I think it may have to do with having the describe and a root test?
> 
> ```js
> describe('test project', () => {
>   it.only('New test', function() {
>     cy.visit('https://www.cypress.io')
> 
>   });
> })
> it('My new test', function() {
> 
> });
> ```
> 
> <img alt="Image" width="1335" height="580" src="https://private-user-images.githubusercontent.com/1271364/472164163-9836876d-2c4d-4ad3-97f0-0e8ea0c98cc0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM4OTI0MDMsIm5iZiI6MTc1Mzg5MjEwMywicGF0aCI6Ii8xMjcxMzY0LzQ3MjE2NDE2My05ODM2ODc2ZC0yYzRkLTRhZDMtOTdmMC0wZThlYTBjOThjYzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDczMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA3MzBUMTYxNTAzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzExMjY0N2MyZDk4NDJiZGFkZGFmOGQ3ZWEyMjdjMzg0ZGFiMDEzOGIxMWM1NWFkOGVjMmQ4ODA0YjkyM2ZiNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.2jwhZKxFop3VfHb8KvMnL_iaerznYczj_jNzox6sYeE">

I fixed this and added a test for it

mabela416 commented on packages/app/src/store/studio-store.ts:
```diff
@@ -232,15 +235,32 @@ export const useStudioStore = defineStore('studioRecorder', {
-        getCypress().runner.setOnlyTestId(this.testId)
+        if (this._wasStudioCreatedTest) {
+          // This test was just created by studio, so we need to set the test id
+          getCypress().runner.setOnlyTestId(this.testId)
```
this doesn't have to be repeated. it could be outside the if since this needs to be set in either scenario

astone123 replied:
good call 👍🏻

mabela416 commented on packages/app/src/store/studio-store.ts:
```diff
@@ -232,15 +235,32 @@ export const useStudioStore = defineStore('studioRecorder', {
       // if this test is the one we created, we can just set the test id
       if (this.newTestLineNumber && test.invocationDetails?.line === this.newTestLineNumber) {
+        this._wasStudioCreatedTest = true
+        this.setTestId(test.id)
```
same here

mabela416 replied:
actually maybe this can be extracted into a function that both these conditions call? since they're both doing the same logic

astone123 replied:
I just moved the condition up to the main if so we only write it once 👍🏻

mabela416 reviewed (changes requested):
I'm getting this error trying to save to a new test created from a suite. If I create a test from the root, I don't get any errors


https://github.com/user-attachments/assets/d0194a3e-9829-4281-9310-6305a9db8d06



<img width="1240" height="211" alt="image" src="https://github.com/user-attachments/assets/04e4f155-748e-4eab-a68a-478bebc50306" />

mabela416 reviewed (approved):
left some comments but otherwise looks good

astone123 committed:
refactors

astone123 committed:
Merge branch 'release/15.0.0' into studio-fix-only

jennifer-shehane commented on packages/app/src/store/studio-store.ts:
```diff
@@ -109,6 +109,7 @@ interface StudioRecorderState {
   showUrlPrompt: boolean
   cloudStudioRequested: boolean
   cloudStudioSessionId?: string
+  _wasStudioCreatedTest: boolean
```
Would probably use present tense here for variables - `_isStudioCreatedTest` - was would imply it no longer has that state?

astone123 replied:
right - updated 👍🏻

jennifer-shehane commented on packages/app/src/store/studio-store.ts:
```diff
@@ -232,15 +235,35 @@ export const useStudioStore = defineStore('studioRecorder', {
+          // this is an existing test - respect .only filtering
+          getCypress().runner.setOnlyTestId(this.testId)
+          getCypress().runner.setIsStudioCreatedTest(false)
+        }
```
This is a bit confusing to read. Perhaps have some logic pulled out:

```js
get shouldBypassOnlyFiltering() {
  return this.newTestLineNumber || this._isStudioCreatedTest
}
```

astone123 replied:
@jennifer-shehane are you looking at the most recent commit? I had made some changes to this

jennifer-shehane commented on packages/app/src/store/studio-store.ts:
```diff
@@ -232,15 +235,35 @@ export const useStudioStore = defineStore('studioRecorder', {
+        getCypress().runner.setIsStudioCreatedTest(true)
+      } else if (this.suiteId) {
+        this.setTestId(test.id)
+        getCypress().runner.setIsStudioCreatedTest(true)
```
I'm not understanding why there's an if/else here if they are setting the exact same thing in each call. If it needs to just meet both conditions to apply, then maybe:

```
const isNewlyCreatedTest = this.newTestLineNumber && test.invocationDetails?.line === this.newTestLineNumber

if (isNewlyCreatedTest || this.suiteId) {
```

astone123 replied:
same here, I think this is a comment on older changes

astone123 committed:
update variable name, consolidate logic

astone123 committed:
Merge branch 'release/15.0.0' into studio-fix-only

astone123 committed:
Merge branch 'release/15.0.0' into studio-fix-only
